releng/jenkins/bot-updates/Jenkinsfile

```diff
@@ -267,22 +417,24 @@ pipeline {
           // publish for each repo
           REPOSITORY_NAMES.split(',').each {
             dir (it) {
-              sh label: "Push changes for ${it}", script: """
-                git push --force origin ${params.TARGET_BRANCH}
-              """
-              
+              env.HAS_CHANGES = sh label: "Evaluate if changes have been made to ${it}", returnStatus: true, script: "git diff ${env.TARGET_BRANCH}..${params.SOURCE_BRANCH} --quiet"
+
+              if ("${env.HAS_CHANGES}" != "0" || params.PUSH_UNCHANGED_BRANCH) {
```

Comment:
In the other stages, you do `git diff-index --quiet HEAD || git commit ...` to avoid no-change-commits. What's different here?